### PR TITLE
Code cleanup on engine initialization functions

### PR DESCRIPTION
**Pull request by Ershany**
Shuffled around the structure to of some of the functions in engine initialization, to increase readability and made the function contain less lines of code.

### CODE REVIEW

File: src/game_initialization/create_engine.cpp [hunk 1]
```diff
@@ -49,24 +49,21 @@ static lg::log_domain log_mp_create_engine("mp/create/engine");
...
-		bool ok = true;
-		for (size_t j = 0; j<str2.size(); j++) {
-			if (std::tolower(str1[i+j]) != std::tolower(str2[j])) {
-				ok = false;
-				break;
+	if (!(str2.size() > str1.size())) {
+		for (size_t i = 0; i < str1.size() - str2.size() + 1; i++) {
+			bool flag = true;
+			for (size_t j = 0; j < str2.size(); j++) {
+				if (std::tolower(str1[i + j]) != std::tolower(str2[j])) {
```
Comment (aginor) [inline src/game_initialization/create_engine.cpp]: indentation is wrong

**Comment by Vultraz** [comment]
Please rebase this PR with master. to get rid of the merge commits.

**Comment by gfgtdf** [comment]
i don't think that the new code is actually cleaner than the old code.

**Comment by CelticMinstrel** [comment]
The old code is cleaner in my opinion. It's better to exit fast rather than introducing extra levels of indentation.

**Comment by aginor** [comment]
I think the new version is harder to read. It's very common practise to exit/return early when your immediate needs are not satisfied so that you do not have to add extra complexity to your core logic.

**Pull request closed without merging**